cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/ActionSteps.java

```diff
@@ -240,19 +231,23 @@ private void handleSerializableSessionContextArgument(Map<String, SerializableSe
     @SuppressWarnings("unchecked")
     private Map<String, Serializable> prepareAndRunPythonAction(
             Map<String, Serializable> callArguments,
-            String pythonScript) {
+            String pythonScript,
+            String operationName) {
 
         if (StringUtils.isNotBlank(pythonScript)) {
-            return runPythonAction(callArguments, pythonScript);
+            return runPythonAction(callArguments, pythonScript, operationName);
         }
 
         throw new RuntimeException("Python script not found in action data");
     }
 
     //we need this method to be synchronized so we will not have multiple scripts run in parallel on the same context
-    private synchronized Map<String, Serializable> runPythonAction(Map<String, Serializable> callArguments,
-                                                             String script) {
+    private synchronized Map<String, Serializable> runPythonAction(
+            Map<String, Serializable> callArguments,
+            String script,
+            String operationName) {
 
+        cleanInterpreter(interpreter);
```

Comment:
changed `cleanInterpreter` location because of this situation:

execute a script that fails on not serializable variable issue -- you can not execute another valid script after since variable remained in the context